pkg/configurator/methods.go

```diff
@@ -123,6 +123,18 @@ func (c *client) GetEnvoyLogLevel() string {
 	return constants.DefaultEnvoyLogLevel
 }
 
+// GetLocalProxyMode returns the local proxy mode
+func (c *client) GetLocalProxyMode() configv1alpha2.LocalProxyMode {
+	proxyConfig := c.getMeshConfig().Spec.Sidecar.LocalProxyMode
+	switch proxyConfig {
+	case configv1alpha2.LocalProxyModeLocalhost, configv1alpha2.LocalProxyModePodIP:
+		return proxyConfig
+	default:
+		// unrecognized proxy mode; return the default: ProxyModeLocalhost
```

Comment:
Since the API rejects any other value up front anyway and the default is defined in the CRD, I have a feeling enforcing the default again here isn't worth it since any other theoretical implementation of this interface (including the mock used in tests) would presumably need to do the same thing which may not be obvious. Then we wouldn't necessarily need to add this method to the interface and `GetMeshConfig().Spec.Sidecar.LocalProxyMode` could be used directly instead.

Reply:
I just followed the approach `envoyLogLevel` did. I added the function `GetLocalProxyMode` because it allows us to explicitly define this behavior on the `Configurator` interface, rather than an implementation detail of `MeshConfig`. You're right though, at the moment, it's not likely this codepath gets hit